illaclient/src/main/java/illarion/client/world/interactive/InteractionManager.java

```diff
@@ -87,24 +87,24 @@ public ItemCount getMovedAmount() {
      */
     public void dropAtContainer(int container, int slot, @Nonnull ItemCount count) {
         if (draggedObject == null) {
-            LOGGER.warn("Dropping to container called without a active dragging operation.");
+            LOGGER.warn("Dropping to container called without an active dragging operation.");
             cancelDragging();
-        }
-
-        ItemContainer itemContainer = World.getPlayer().getContainer(container);
-        if (itemContainer == null) {
-            LOGGER.error("Container a item was dropped at was not found.");
-            return;
-        }
+        }else { // If cancelDragging() didn't stop the sequence, don't do anything anyway
+            ItemContainer itemContainer = World.getPlayer().getContainer(container);
+            if (itemContainer == null) { // If the container is not valid
+                LOGGER.error("Container an item was dropped at was not found.");
+            }else {
 
-        try {
-            InteractiveContainerSlot targetSlot = itemContainer.getSlot(slot).getInteractive();
-            draggedObject.dragTo(targetSlot, count);
-        } catch (@Nonnull IndexOutOfBoundsException ex) {
-            LOGGER.error("Tried to drop a item at a container slot that does not exist.", ex);
-        } finally {
-            cancelDragging();
-        }
+                try {
+                    InteractiveContainerSlot targetSlot = itemContainer.getSlot(slot).getInteractive();
+                    draggedObject.dragTo(targetSlot, count); // Enter the contents into the slot
+                } catch (@Nonnull IndexOutOfBoundsException ex) {
```

Comment:
I consider catching the IndexOutOfBoundsException as very bad practise.
We should rather try to find the reason why the wrong slot is requested.